R/merge.R

```diff
@@ -121,16 +130,24 @@
 
 #' @importFrom S4Vectors SimpleList
 #' @importFrom scuttle summarizeAssayByGroup
-.merge_cols <- function(x, f, archetype = 1L, ...){
+.merge_cols <- function(x, f, archetype = 1L, na.rm = FALSE, ...){
     # input check
+    if(!.is_a_bool(na.rm)){
+        stop("'na.rm' must be TRUE or FALSE.", call. = FALSE)
+    }
+    # This is done otherwise we lose NA values
+    if ( na.rm ) {
+        f[is.na(f)] <- "NA"
+    }
     if(is.character(f) && length(f)==1 && f %in% colnames(rowData(x))){
         f <- factor(as.character(rowData(x)[, f]))
     }
     else if(is.character(f) && length(f)==1 && f %in% colnames(colData(x))){
+        f[is.na(f)] <- "NA"
         f <- factor(as.character(colData(x)[, f]))
-    } else 
+    } else
     {
-        f <- .norm_f(ncol(x), f, "columns")  
+        f <- .norm_f(ncol(x), f, "columns")
     }
```

Comment:
Modify this also similarly to .merge_rows

Reply:
ACK